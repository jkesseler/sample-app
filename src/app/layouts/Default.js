import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Paper } from '@material-ui/core';
import routes from '../routes';

import Header from '../components/Header';

export const Default = ({ children }) => (
  <Paper
    elevation={0}
    style={{
      padding: 0,
      margin: 0,
      backgroundColor: '#fafafa',
    }}
  >
    <Header menuItems={routes}>Todos</Header>
    { children }
  </Paper>
);

Default.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default memo(Default);
