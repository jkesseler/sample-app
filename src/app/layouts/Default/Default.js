import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Paper, Typography } from '@material-ui/core';
import Header from '~ui/Header';
import routes from '../../routes';

import styles from './Default.module.css';

export const Default = ({ children, title, displayFooter = true }) => (
  <div className={styles.main}>
    <Paper
      elevation={0}
      className={styles.content}
    >
      <Header menuItems={routes}>{title}</Header>
      { children }
    </Paper>

    {displayFooter && (
      <footer className={styles.footer}>
        <Typography variant="h6" align="center" gutterBottom>
          Footer
        </Typography>
        <Typography variant="subtitle1" align="center" color="textSecondary" component="p">
          Something here to give the footer a purpose!
        </Typography>
      </footer>
    )}
  </div>
);

Default.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  title: PropTypes.string.isRequired,
  displayFooter: PropTypes.bool,
};

Default.defaultProps = {
  displayFooter: false,
};

export default memo(Default);
