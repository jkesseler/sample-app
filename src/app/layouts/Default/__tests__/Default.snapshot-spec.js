import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Default } from '../Default';

const Child = () => (<span>fake child</span>);

describe('Default', () => {
  it('renders with footer', () => {
    const component = shallow(<Default title="Title" displayFooter><Child /></Default>);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders without footer', () => {
    const component = shallow(<Default title="Title" displayFooter={false}><Child /></Default>);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
