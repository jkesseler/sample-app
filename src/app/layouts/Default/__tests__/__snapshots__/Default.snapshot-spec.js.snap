// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base renders without crashing 1`] = `
<div
  className="main"
>
  <WithStyles(Paper)
    className="content"
    elevation={0}
  >
    <React.Memo
      menuItems={
        Array [
          Object {
            "component": Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            },
            "exact": true,
            "layout": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            },
            "path": "/",
            "title": "Home",
          },
          Object {
            "component": Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            },
            "exact": true,
            "layout": undefined,
            "path": "/todos",
            "title": "Todos",
          },
        ]
      }
    >
      Todos
    </React.Memo>
    <Child />
  </WithStyles(Paper)>
  <footer
    className="footer"
  >
    <WithStyles(Typography)
      align="center"
      gutterBottom={true}
      variant="h6"
    >
      Footer
    </WithStyles(Typography)>
    <WithStyles(Typography)
      align="center"
      color="textSecondary"
      component="p"
      variant="subtitle1"
    >
      Something here to give the footer a purpose!
    </WithStyles(Typography)>
  </footer>
</div>
`;
